src/deepsparse/transformers/pipelines/text_generation.py

```diff
@@ -497,33 +531,29 @@ def process_engine_outputs(
         :param engine_outputs: the outputs from the engine
         :return: the output schema for the pipeline
         """
-        generated_tokens, generated_logits, finished_reason, *debug = engine_outputs
-        finished_reason = [f[0] for f in finished_reason]
 
+        prompts = kwargs.get("prompts")
+        streaming = kwargs.get("streaming")
+
+        if streaming:
+            return self._stream_engine_outputs(engine_outputs, prompts, kwargs)
+
+        generated_tokens, generated_logits, finished_reason, *debug = list(
+            *engine_outputs
+        )
         sequences = self.tokenizer.batch_decode(
             generated_tokens, skip_special_tokens=True
         )
-        num_preds = kwargs.get("num_generated_predictions", 1)
-        prompts = kwargs.get("prompts")
-
-        def _create_generated_text_output(
-            sequence: str,
-            finish_reason: FinishReason,
-            logits: Optional[numpy.array] = None,
-        ):
-            return GeneratedText(
-                text=sequence,
-                score=logits,
-                finished=True,
-                finished_reason=finish_reason.value,
-            )
 
         logits = generated_logits if kwargs.get("return_logits") else None
 
+        num_preds = kwargs.get("num_generated_predictions", 1)
+        finished_reason = [f[0] for f in finished_reason]
```

Comment:
We'll always have a 0th index right?

Reply:
yes because the engine_forward function always wraps the outputs into a list